### PR TITLE
Consumer, Producer and Group Consume will optionally poll for new partitions and react accordingly

### PR DESCRIPTION
**Pull request by bbalser**

### CODE REVIEW

File: lib/elsa/supervisor.ex [hunk 1]
```diff
@@ -87,6 +91,8 @@ defmodule Elsa.Supervisor do
 
   * `:handler_init_args` - Optional. Any args to be passed to init function in handler module.
 
+  * `:poll` - Optional, If set to number of milliseconds, will poll for new partitions and startup consumers on the fly.
```
Comment (jdenen) [inline lib/elsa/supervisor.ex]:
```suggestion
  * `:poll` - Optional. If set to number of milliseconds, will poll for new partitions and startup consumers on the fly.
```

File: lib/elsa/consumer/initializer.ex [hunk 2]
```diff
@@ -23,6 +23,8 @@ defmodule Elsa.Consumer.Initializer do
   end
 
   defp configure_topic(topic, registry, brod_client, config) do
+    :brod_client.get_metadata(brod_client, topic)
```
Comment (jdenen) [inline lib/elsa/consumer/initializer.ex]: Wrap this in a private function to better communicate that it updates global state from `brod`?

File: lib/elsa/util.ex [hunk 3]
```diff
@@ -105,8 +105,8 @@ defmodule Elsa.Util do
   Return the number of partitions for a given topic. Bypasses the need for a persistent client
   for lighter weight interactions from one-off calls.
   """
-  @spec partition_count(keyword(), String.t()) :: integer()
-  def partition_count(endpoints, topic) do
+  @spec partition_count(keyword | Elsa.connection() | pid, String.t()) :: integer()
```
Comment (jeffgrunewald) [inline lib/elsa/util.ex]: why are you handling a single topic here when the first arguments is a keyword list of endpoints, but the one below when the first argument is an atom or a pid looks to take a list of topics?

Comment (jeffgrunewald) [reply lib/elsa/util.ex]: line 110 vs 120

Comment (bbalser) [reply lib/elsa/util.ex]: :brod.get_metadata take a list of topics, :brod_client.get_metadata only takes a single topic.

Comment (jeffgrunewald) [reply lib/elsa/util.ex]: Why the two different strategies?

Comment (bbalser) [reply lib/elsa/util.ex]: the new one uses an existing connection rather then establishing a new connection to just get metadata.

Comment (jeffgrunewald) [reply lib/elsa/util.ex]: Got it; didn't realize the other one would require a full extra roundtrip to the cluster. Assuming that isn't essentially required to ensure up-to-date accuracy (the existing connection won't have a potentially inaccurate cache of info) that makes sense to me.

**Pull request merged**